### ISSUE TITLE
kubernetes client initialization failed

### ISSUE THREAD

**Issue by luislhl** [fluxcd/flux2#2817]
### Describe the bug

After upgrading to v0.31.0, I'm getting this error when trying to run any Flux command, like `flux version` or `flux check`.

> kubernetes client initialization failed: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"

I can output the client version:

> flux --version
flux version 0.31.0

This is my kubectl version:

> ✗ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.15-eks-a64ea69", GitCommit:"03450cdabfc4162d4e447e6d8c5037efe6d29742", GitTreeState:"clean", BuildDate:"2022-05-12T18:44:04Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}

### Steps to reproduce

What I did was:
1. Upgrade my Flux installation to v0.31.0 (I have automated the upgrades with Github ACtions, so I don't have to run the CLI in my machine for the upgrade)
2. Downloaded the new Flux CLI in my machine
3. Tried to run any command with the new CLI

### Expected behavior

I expected the CLI to work as before

### OS / Distro

Fedora 33

### Flux version

v0.31.0

### Flux check

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefanprodan** [comment]
Please see https://github.com/aws/aws-cli/issues/6920

**TLDR**
The `client.authentication.k8s.io/v1alpha1` has been deprecated and removed from Kubernetes 1.24. Update your kubectl and you'll see the same error.

To fix this you need to update your AWS CLI to the latest version and regenerate the kubeconfig with:

```sh
aws eks update-kubeconfig --name ${EKS_CLUSTER_NAME} --region ${REGION}
```

**Comment by rajughoshdev** [comment]
Thanks, @stefanprodan I just looking for the solution. And you just update it a few min back.

**Comment by infakt-HNP** [comment]
If you use brew to install kubectl, kubectx and aws (aws-cli), you can temp. downgrade and lock to 1.22 version using:
```
brew uninstall --ignore-dependencies kubectl
brew install kubectl@1.22
brew pin kubectl@1.22
```

**Comment by stefanprodan** [comment]
@infakt-HNP The latest Flux CLI is build with Kubernetes 1.24 packages, the only way to fix this for Flux is to move forward and update the kubeConfig to `client.authentication.k8s.io/v1beta1`.

**Comment by luislhl** [comment]
Thanks, @stefanprodan , worked like a charm.

I'll close the issue

**Comment by cchukwu** [comment]
@[stefanprodan](https://github.com/stefanprodan) this resolved it for me as well
